File: scripts/convert_csv_to_yaml.js
const fs = require('fs');
const yaml = require('js-yaml');
const csv = require('csv-parser');

const manufacturersByCategory = {};
const results = [];

// CSVファイルを読み込む
fs.createReadStream('data/Makers_from_webflow_20251206.csv')
  .pipe(csv())
  .on('data', (row) => {
    // Display=falseの場合はスキップ
    if (row['Display']?.toLowerCase() !== 'true') {
      return;
    }

    const makerName = row['Maker Name'];
    const makerNameEn = row['Maker Name (EN)'];
    const hasPartnership = row['Trade'] === '取引';
    const link = row['Link'] || null;

    // KindsとKinds2を統合してカテゴリを決定
    const categories = [];
    if (row['Kinds'] && row['Kinds'] !== '取引' && row['Kinds'] !== '' && !row['Kinds'].startsWith('http')) {
      categories.push(row['Kinds']);
    }

    if (row['Kinds2'] && row['Kinds2'] !== '取引' && row['Kinds2'] !== '' && !row['Kinds2'].startsWith('http')) {
      categories.push(row['Kinds2']);
    }

    // カテゴリがない場合は「その他」
    if (categories.length === 0) {
      categories.push('その他');
    }

    // 油圧・空圧機器と油空圧・液体機器を統合
    const normalizedCategories = categories.map(cat =>
      (cat === '油圧・空圧機器' || cat === '油空圧・液体機器') ? '油空圧・液体機器' : cat
    );

    // 各カテゴリに追加
    normalizedCategories.forEach(category => {
      if (!manufacturersByCategory[category]) {
        manufacturersByCategory[category] = [];
      }

      const manufacturer = {
        name: makerName,
        name_en: makerNameEn,
        has_partnership: hasPartnership
      };

      if (link) {
        manufacturer.link = link;
      }

      manufacturersByCategory[category].push(manufacturer);
    });
  })
  .on('end', () => {
    // カテゴリ順序を定義
    const categoryOrder = [
      '作業工具',
      '荷役運搬機器',
      '測定機器',
      '切削工具',
      '溶接関連',
      '塗装',
      '油空圧・液体機器',
      'ボルト・ナット',
      '安全具',
      '仮設機器',
      'ケミカル品',
      '電気',
      '橋梁・土木',
      '環境機器',
      '軸受（ベアリング）',
      '海外メーカー',
      'その他'
    ];

    // YAML出力用のデータ構造
    const outputData = {
      manufacturers: {
        categories: []
      }
    };

    // カテゴリ順に並べる
    categoryOrder.forEach(categoryName => {
      if (manufacturersByCategory[categoryName]) {
        const companies = manufacturersByCategory[categoryName];
        // 会社名でソート
        companies.sort((a, b) => a.name_en.localeCompare(b.name_en));

        outputData.manufacturers.categories.push({
          name: categoryName,
          companies: companies
        });
      }
    });

    // YAMLファイルに出力
    const yamlStr = yaml.dump(outputData, {
      noRefs: true,
      lineWidth: -1,
      sortKeys: false
    });

    fs.writeFileSync('content/manufacturers_new.yaml', yamlStr, 'utf8');

    console.log('✅ YAMLファイルを生成しました: content/manufacturers_new.yaml');
    console.log(`📊 カテゴリ数: ${outputData.manufacturers.categories.length}`);
    const totalCompanies = outputData.manufacturers.categories.reduce((sum, cat) => sum + cat.companies.length, 0);
    console.log(`📊 メーカー数: ${totalCompanies}`);
  });

File: scripts/convert_csv_to_yaml.py
#!/usr/bin/env python3
"""
Convert CSV manufacturer data to YAML format
Filters by Display=true and groups by Kinds and Kinds2 categories
"""

import csv
import json
from collections import defaultdict
from pathlib import Path

def convert_csv_to_yaml(csv_path, yaml_path):
    # Read CSV file
    manufacturers_by_category = defaultdict(list)

    with open(csv_path, 'r', encoding='utf-8') as f:
        reader = csv.DictReader(f)

        for row in reader:
            # Filter: Only Display=true
            if row['Display'].strip().lower() != 'true':
                continue

            # Determine category (Kinds2 takes priority, then Kinds)
            category = row['Kinds2'].strip() if row['Kinds2'].strip() else row['Kinds'].strip()

            if not category:
                print(f"Warning: No category for {row['Maker Name']}")
                continue

            # Create manufacturer entry
            manufacturer = {
                'name': row['Maker Name'].strip(),
                'name_en': row['Maker Name (EN)'].strip(),
                'link': row['Link'].strip() if row['Link'].strip() else '',
                'has_partnership': row['Trade'].strip() == '取引'
            }

            manufacturers_by_category[category].append(manufacturer)

    # Sort categories and manufacturers
    categories = []
    for category_name in sorted(manufacturers_by_category.keys()):
        companies = manufacturers_by_category[category_name]
        # Sort companies by name
        companies.sort(key=lambda x: x['name'])

        categories.append({
            'name': category_name,
            'companies': companies
        })

    # Write YAML file manually
    with open(yaml_path, 'w', encoding='utf-8') as f:
        f.write('manufacturers:\n')
        f.write('  categories:\n')
        
        for category in categories:
            f.write(f'    - name: {category["name"]}\n')
            f.write('      companies:\n')
            
            for company in category['companies']:
                f.write(f'        - name: {company["name"]}\n')
                f.write(f'          name_en: {company["name_en"]}\n')
                f.write(f'          has_partnership: {str(company["has_partnership"]).lower()}\n')
                if company['link']:
                    f.write(f'          link: {company["link"]}\n')

    # Print summary
    total_manufacturers = sum(len(cat['companies']) for cat in categories)
    print(f"Successfully converted {total_manufacturers} manufacturers into {len(categories)} categories")
    print(f"Output written to: {yaml_path}")

    # Print category summary
    print("\nCategories:")
    for cat in categories:
        print(f"  - {cat['name']}: {len(cat['companies'])} companies")

if __name__ == '__main__':
    csv_path = Path(__file__).parent.parent / 'data' / 'Makers_from_webflow_20251206.csv'
    yaml_path = Path(__file__).parent.parent / 'content' / 'manufacturers.yaml'

    convert_csv_to_yaml(csv_path, yaml_path)
